src/App.tsx

```diff
@@ -42,16 +45,52 @@ const App: React.FC = (): ReactElement => {
 
   function parseAttemptHistoryFromFile(fileObject: any): Attempt[] {
     const history: Attempt[] = fileObject?.Run?.AttemptHistory?.[0]?.Attempt?.flatMap(
-      ({ $, RealTime }: any) => {
-        return $?.id && RealTime?.[0]
-          ? { x: Number($.id), y: moment.duration(RealTime[0]).as('milliseconds') }
+      ({ id, RealTime }: any) => {
+        return id && RealTime
+          ? { x: Number(id), y: moment.duration(RealTime).as('milliseconds') }
           : []
       }
     )
 
     return history
   }
 
+  function parseSegmentsFromFile(fileObject: any): Segment[] {
+    const segmentData: Segment[] = fileObject?.Run?.Segments?.[0]?.Segment?.flatMap(
+      ({ Name, SplitTimes, BestSegmentTime, SegmentHistory }: any) => {
+        let name: string | null = null
+        if (Name?.[0]) {
```

Comment:
name = Name?.[0] || null 

? may e works

Reply:
does `let name: string? = Name?.[0]` work? (with name nullable and the optional chain)

Reply:
`const name: string | null = Name?.[0]` works.  I get an error if I try to use string? for the type